Cattac/Cattac/GameEngine.swift

```diff
@@ -150,13 +161,22 @@ class GameEngine {
         }
     }
     
+    /// Releases all resources associated with this game, called
+    /// when game has ended.
     func end() {
         for ref in movementWatchers.values {
             ref.removeAllObservers()
         }
     }
     
+    /// Trigger state advancement in game engine.
     private func nextState() {
```

Comment:
Yea why didn't I think of that. lol okay 9cfe93a5255592913388d5ad3d1a9d0e078c63af